tools/syntax.txt

```diff
@@ -1096,6 +1110,40 @@ LetPortListSyntax portList
 EqualsValueClauseSyntax initializer
 token semi
 
+AssertionVariableDeclarationSyntax kind=AssertionVariableDeclaration
+DataTypeSyntax type
+separated_list<VariableDeclaratorSyntax> assignments
+token semi
+
+PropertyLvarPortDirectionSyntax kind=PropertyLvarPortDirection
+token input
+
+PropertyLocalPortSyntax kind=PropertyLocalPort
+token local
+PropertyLvarPortDirectionSyntax direction
```

Comment:
You don't need a whole separate syntax node type for optional tokens; a default constructed Token() represents one that is empty.